File: backend/app/schemas/lead.py
from pydantic import BaseModel, EmailStr
from datetime import datetime


class LeadBase(BaseModel):
    first_name: str
    last_name: str
    email: EmailStr
    company: str
    job_title: str | None = None
    status: str = "new"
    value: float = 0.0
    journey_id: str
    stage_id: str | None = None
    persona_id: str | None = None


class LeadCreate(LeadBase):
    pass


class LeadUpdate(BaseModel):
    first_name: str | None = None
    last_name: str | None = None
    email: EmailStr | None = None
    company: str | None = None
    job_title: str | None = None
    status: str | None = None
    value: float | None = None
    journey_id: str | None = None
    stage_id: str | None = None
    persona_id: str | None = None


class LeadStageResponse(BaseModel):
    id: str
    name: str
    icon: str
    color: str

    model_config = {"from_attributes": True}


class LeadResponse(LeadBase):
    id: str
    created_at: datetime
    updated_at: datetime
    stage: LeadStageResponse | None = None

    model_config = {"from_attributes": True}

File: frontend/src/types/lead.ts
import { Stage } from "./journey";

export interface Lead {
  id: string;
  first_name: string;
  last_name: string;
  email: string;
  company: string;
  job_title?: string;
  status: string;
  value: number;
  journey_id: string;
  stage_id?: string;
  persona_id?: string;
  stage?: Stage;
  created_at: string;
  updated_at: string;
}
